File: twitter.js
const { email, id, password } = require('./private.js');
const { login } = require('./twitter_login.js');
const { Builder, By, until, Key } = require('selenium-webdriver');
const fs = require('fs');
// console.log(Key)
// return
const nickName = "mizuki_i_l";

// 트위터 크롤링
const start =  async () => {
	let driver = await new Builder('./chromedriver').forBrowser('chrome').build();
	await driver.get('https://twitter.com/i/flow/login');
	await driver.manage().window().maximize();

	// 메인 브라우저 값 저장
	const defaultHandle = await driver.getWindowHandle();
	
	// 트위터 로그인
	login(driver, email, id, password);

	// 트윗 내용 크롤링
	await driver.wait(until.elementLocated(By.css('[data-testid="cellInnerDiv"]')));
	await driver.get(`https://twitter.com/${nickName}`);
	
	// 트윗 링크 리스트
	let tweetsLink = [];
	
	try {
		while(1) {
			let replyCount = 0;
			await driver.wait(until.elementLocated(By.css('[data-testid="cellInnerDiv"]')));
			await driver.findElement(By.css('[data-testid=cellInnerDiv] [data-testid=app-text-transition-container]')).getText().then(function(text) {
				replyCount = Number(text);
			});
			
			// 리플 갯수 1개 이상에서만 실행
			if (0 < replyCount) {
				const tweetLinkTag = await driver.findElements(By.css('[data-testid=cellInnerDiv]:first-child a[role=link]'));
				const urlPatten = new RegExp(`${nickName}/status/.*`);
				const urlPatten2 = new RegExp(`photo`);
				for (let i = 0; i < tweetLinkTag.length; i++) {
					let tweetLink = await tweetLinkTag[i].getAttribute('href');
					if (urlPatten.test(tweetLink) && !urlPatten2.test(tweetLink)) {
						tweetsLink.push(tweetLink);
						console.log(tweetsLink)
					}
				}
			}
		
			//트윗 block 삭제
			await driver.wait(until.elementLocated(By.css('[data-testid="cellInnerDiv"]')));
			await driver.executeScript(`
				document.querySelector('[data-testid="cellInnerDiv"]').remove();
			`);
			await driver.sleep(100);
		}
	} catch (error) {

		tweetsLink.forEach()
		fs.writeFile('links.txt', "tweetsLink", 'utf8', function(error){
			console.log('write end!')
		});
	}

	// await driver.quit();
};

const userAccess =  async (nickName) => {
	await driver.wait(until.elementLocated(By.css('[data-testid="cellInnerDiv"]')));
	await driver.get(`https://twitter.com/${nickName}`);
}

(async ()=>{
	await start();
})();

File: twitter.py
import chromedriver_autoinstaller
from selenium import webdriver
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.common.keys import Keys
from time import sleep
from private import *

# chrome driver를 자동으로 설치함
chromedriver_autoinstaller.install() 

options = webdriver.ChromeOptions() # Browser 세팅하기
options.add_argument('lang=ko_KR') # 사용언어 한국어
# options.add_argument('disable-gpu') # 하드웨어 가속 안함
options.add_argument('--start-maximized') # 하드웨어 가속 안함
# options.add_argument('headless') # 창 숨기기

# 브라우저 세팅
driver = webdriver.Chrome(options=options)

# 브라우저에 URL 호출하기
driver.get(url='https://twitter.com/i/flow/login')
driver.implicitly_wait(10)

# 이메일
username_field = driver.find_element_by_name("text")
username_field.send_keys(email)
username_field.send_keys(Keys.RETURN)

# 아이디
username_field2 = driver.find_element_by_name("text")
username_field2.send_keys(id)
username_field2.send_keys(Keys.RETURN)

# 비밀번호
password_field = driver.find_element_by_name("password")
password_field.send_keys(password)
password_field.send_keys(Keys.RETURN)

# 사용자검색
search_field = driver.find_element_by_xpath('//*[@id="react-root"]/div/div/div[2]/main/div/div/div/div[2]/div/div[2]/div/div/div/div[1]/div/div/div/form/div[1]/div/div/div/label/div[2]/div/input')
search_field.send_keys("mizuki_i_l")
sleep(2)
search_field.send_keys(Keys.ARROW_DOWN)
search_field.send_keys(Keys.ARROW_DOWN)
search_field.send_keys(Keys.RETURN)

sleep(2)

# 블럭클릭
tweet_block = driver.find_element_by_css_selector("[data-testid=cellInnerDiv]")
ActionChains(driver).key_down(Keys.CONTROL).click(tweet_block).key_up(Keys.CONTROL).perform()

# 블럭삭제
# js_removeBlock = "var target = document.querySelector('[data-testid=cellInnerDiv]');target.parentNode.removeChild(target)"
# driver.execute_script(js_removeBlock)

sleep(600)
# 브라우저 탭 닫기
driver.close()
# 브라우저 종료하기 (탭 모두 종료)
driver.quit()